q2_diversity/_alpha.py

```diff
@@ -5,3 +5,116 @@
 #
 # The full license is in the file LICENSE, distributed with this software.
 # ----------------------------------------------------------------------------
+
+import os
+
+import biom
+import scipy
+import skbio
+import qiime
+import pandas as pd
+import seaborn as sns
+import skbio.diversity
+from statsmodels.sandbox.stats.multicomp import multipletests
+
+
+# We should consider moving these functions to scikit-bio. They're part of
+# the private API here for now.
+def phylogenetic_metrics():
+    return {'faith_pd'}
+
+
+def non_phylogenetic_metrics():
+    return {'ace', 'chao1', 'chao1_ci', 'berger_parker_d', 'brillouin_d',
+            'dominance', 'doubles', 'enspie', 'esty_ci', 'fisher_alpha',
+            'goods_coverage', 'heip_e', 'kempton_taylor_q', 'margalef',
+            'mcintosh_d', 'mcintosh_e', 'menhinick', 'michaelis_menten_fit',
+            'observed_otus', 'osd', 'pielou_e', 'robbins', 'shannon',
+            'simpson', 'simpson_e', 'singles', 'strong', 'gini_index',
+            'lladser_pe', 'lladser_ci'}
+
+
+def alpha_phylogenetic(table: biom.Table, phylogeny: skbio.TreeNode,
+                       metric: str) -> pd.Series:
+    if metric not in phylogenetic_metrics():
+        raise ValueError("Unknown phylogenetic metric: %s" % metric)
+
+    counts = table.matrix_data.toarray().astype(int).T
+    sample_ids = table.ids(axis='sample')
+    feature_ids = table.ids(axis='observation')
+
+    result = skbio.diversity.alpha_diversity(
+                metric=metric,
+                counts=counts,
+                ids=sample_ids,
+                otu_ids=feature_ids,
+                tree=phylogeny)
+    result.name = metric
```

Comment:
Yeah, this is much better (let's us use the metric name in the boxplots too, which is nice). Thanks for the suggestion.